frontend/src/components/Button.module.css

```diff
@@ -0,0 +1,29 @@
+.button {
+  width: auto;
+  height: 2.25rem;
+  padding: 0 0.75rem;
+  border: none;
+  border-radius: 0.5rem;
+  font: var(--font-label);
+  cursor: pointer;
+}
```

Comment:
is rem always preferred to px/other, non-relative units?

Reply:
Yep, units of `rem` and sometimes `em` are generally preferable in CSS. `em` refers to the "em size" of the current font, which is basically a number proportional to the font size. `rem` refers to the em size at the root of the DOM, which is set by the browser (typically 16px - this means if some text doesn't have a font size set or inherited then it defaults to 16px).

The main reason these units are preferable is that users may change the rem size, say if they're visually impaired and want text to be bigger. We'd likely want most, if not all, elements on the page to scale appropriately with that, including non-text things like padding, and we can be more certain of that behavior by using `rem` as much as possible instead of units like `px` or `pt`. This also applies when the user zooms in or out via the browser. Nowadays, I think the major browsers actually scale other units too when you change the zoom level, but we shouldn't assume that will happen.

In most cases I usually default to `rem`, and use `em` for things that do depend on the current font size (such as the top margin of a heading - for smaller headings, we might want smaller font and smaller margin). For purposes of implementing a design, you can assume that `1rem` equals `16px` and calculate from there.

Reply:
See https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#numbers_lengths_and_percentages for more info

Reply:
Thanks for the detailed answer! I've always been aware of rem/em but have historically used px. Hadn't considered the end user changing their rem size. I love learning the reasoning behind this stuff.